user_data/hyperopts/sample_hyperopt.py

```diff
@@ -167,42 +172,6 @@ def sell_indicator_space() -> List[Dimension]:
                          'sell-sar_reversal'], name='sell-trigger')
         ]
 
-    @staticmethod
-    def generate_roi_table(params: Dict) -> Dict[int, float]:
-        """
-        Generate the ROI table that will be used by Hyperopt
-        """
-        roi_table = {}
-        roi_table[0] = params['roi_p1'] + params['roi_p2'] + params['roi_p3']
-        roi_table[params['roi_t3']] = params['roi_p1'] + params['roi_p2']
-        roi_table[params['roi_t3'] + params['roi_t2']] = params['roi_p1']
-        roi_table[params['roi_t3'] + params['roi_t2'] + params['roi_t1']] = 0
-
-        return roi_table
-
-    @staticmethod
-    def stoploss_space() -> List[Dimension]:
-        """
-        Stoploss Value to search
-        """
-        return [
-            Real(-0.5, -0.02, name='stoploss'),
-        ]
-
-    @staticmethod
-    def roi_space() -> List[Dimension]:
-        """
-        Values to search for each ROI steps
-        """
-        return [
-            Integer(10, 120, name='roi_t1'),
-            Integer(10, 60, name='roi_t2'),
-            Integer(10, 40, name='roi_t3'),
-            Real(0.01, 0.04, name='roi_p1'),
-            Real(0.01, 0.07, name='roi_p2'),
-            Real(0.01, 0.20, name='roi_p3'),
```

Comment:
This creates one problem - there is no template on how to do this anymore (which is what `sample_hyperopt`used to be here for)

As a user, i would not want do digg through the parent class (which i don't know where it is) to find how the interface for this looks like.

Maybe adding the template for each of the functions to the hyperopt documentation will suffice...?

Reply:
Well, I agree.
I just do not want to add those samples right in the hyperopt.md (and keep it in the sample_hyperopt.md) -- I suppose overriding these methods will be rarely required and is not absolutely needed for a novice user reading the docs. But you are right -- for a more advanced user there should be an explicit example.

In principle, there can be the new "Advanced Freqtrade Techniques" part be added to the docs, with potential content:
* details and examples of using multiple configs
* how to run properly multiple instances of the bot (with common parts of config, with diff. databases)
* _how to override these hyperopt methods. (this issue)_
* how to run freqtrade as a system service
* more on hyperopting, etc.

In the meantime, I could create the `sample_advanced_hyperopt.py` (naming?) with those methods (a copy of current (old) sample_hyperopt.py with some changes in the comments). The "Advanced Freqtrade Techniques" can later contain quotes from it and refer to it.

Reply:
>  I suppose overriding these methods will be rarely required 

this may be true for `stoploss_space()` and `generate_roi_table()` (especially this one!!), however `roi_space()` needs to be changed depending on the timeframe, so every other user will probably need to override this.

Assuming you optimize on 1h candles, in which case our defaults make 0 sense, since 2 of the 3 options are < 1 candle size.
i guess your assumption is similar to mine - using ~5m candles - but it's dangerous since i see many users seem to use 30m, 1h or even longer timeframes, in which case this method is completely unsuitable. 

The documentation does require a general overhaul / reorganisation ... some parts are duplicated all over the place, and others are missing completely (but that shall not be part of this PR).

Reply:
Agree (for `roi_space` on long timeframes), will think about this point...

Reply:
in hyperopt, or in general the roi dict?

Reply:
both. in hyperopt it's a set of ranges, in the strategy -- just a point from that parameters hyperspace...

For this PR, I'm going to add a new `AdvancedSampleHyperOpts` (in the new `advanced_sample_hyperopt.py`) class, which keeps the `stoploss_space()`, `generate_roi_table()` and `roi_space()` as samples of how they can be overridden, with appropriate comments. The `stoploss_space()` sample will shorten the default range of the stoploss hyperspace, `generate_roi_table()` will be same with some extra notes on when this method should be overridden, and a 'scaled' version of the 'roi_space()' which is appropriate for 1h and 1d timeframes. I know how to scale minutes in the roi tables (the roi_tN values --  by multiplying the default ranges to the timeframe period), but thinking on how to scale the ranges of the roi values (the roi_pN values), what can be appropriate limits for their ranges in the case of long timeframes...

Reply:
well that's the point - it's intended to be adjusted to the strategy / timeframe you're using.

I wouldn't change it from minutes to candles though, as i may want to change ROI it mid-candle (hourly candles, lower ROI after 90 minutes since i know that it either wins immediately, or not at all ...).